### PR TITLE
[master < T0880-MG] Add information on show replicas to express how uptodate is replica

### PR DESCRIPTION
**Pull request by 42jeremy**
- [x] Check, and update documentation if necessary
- [x] Update [changelog](https://docs.memgraph.com/memgraph/changelog)

Documentation [PR](https://github.com/memgraph/docs/pull/439)

### CODE REVIEW

File: src/storage/v2/replication/replication_client.cpp [hunk 1]
```diff
@@ -537,6 +537,33 @@ std::vector<Storage::ReplicationClient::RecoveryStep> Storage::ReplicationClient
...
+  try {
+    auto stream{rpc_client_->Stream<replication::TimestampRpc>()};
+    const auto response = stream.AwaitResponse();
+    const auto is_success = response.success;
+    if (!is_success) {
+      replica_state_.store(replication::ReplicaState::INVALID);
+      HandleRpcFailure();
+    }
+    auto main_time_stamp = storage_->last_commit_timestamp_.load();
+    info.current_timestamp = response.current_commit_timestamp;
```
Comment (antaljanosbenjamin) [inline src/storage/v2/replication/replication_client.cpp:554]:
```suggestion
    info.current_timestamp = main_time_stamp;
```
```suggestion
    info.current_timestamp = response.current_commit_timestamp;
```

File: src/query/interpreter.cpp [hunk 2]
```diff
@@ -512,7 +518,8 @@ Callback HandleReplicationQuery(ReplicationQuery *repl_query, const Parameters &
     }
 
     case ReplicationQuery::Action::SHOW_REPLICAS: {
-      callback.header = {"name", "socket_address", "sync_mode", "timeout", "state"};
+      callback.header = {"name",    "socket_address",    "sync_mode",
+                         "timeout", "current_timestamp", "number_of_timestamp_behind_master", "state"};
```
Comment (antaljanosbenjamin) [inline src/query/interpreter.cpp:522]:
```suggestion
                         "timeout", "current_timestamp_of_replica", "number_of_timestamp_behind_master", "state"};
```

As you suggested on slack, we should rename this to make it more clear to users.

**Review by antaljanosbenjamin — CHANGES_REQUESTED** [review]

File: src/storage/v2/replication/replication_client.cpp (hunk 1 above)
Comment (42jeremy) [reply src/storage/v2/replication/replication_client.cpp:554]: Original idea was to have info.current_timestamp containing the current timestamp of the replica (that returned the response). I think I disagree with the suggestion here.
(Note: attribute will be renamed for clarity as `current_timestamp` is definitely confusing)

**Review by antaljanosbenjamin — APPROVED** [review]

File: src/storage/v2/replication/replication_client.cpp [hunk 3]
```diff
@@ -537,6 +537,33 @@ std::vector<Storage::ReplicationClient::RecoveryStep> Storage::ReplicationClient
...
+    const auto is_success = response.success;
+    if (!is_success) {
+      replica_state_.store(replication::ReplicaState::INVALID);
+      HandleRpcFailure();
+    }
+    auto main_time_stamp = storage_->last_commit_timestamp_.load();
+    info.current_timestamp_of_replica = response.current_commit_timestamp;
+    info.current_number_of_timestamp_behind_master = response.current_commit_timestamp - main_time_stamp;
+  } catch (const rpc::RpcFailedException &) {
+    {
```
Comment (kostasrim) [inline src/storage/v2/replication/replication_client.cpp:557]: Why the extra scope?

Comment (42jeremy) [reply src/storage/v2/replication/replication_client.cpp:557]: Nice catch there!

**Review by kostasrim — APPROVED** [review]

**Pull request merged**